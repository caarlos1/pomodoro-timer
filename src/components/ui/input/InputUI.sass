.input-text
    display: flex
    flex-direction: column
    width: 100%

    .label
        font-size: 14px
        margin-bottom: 5px

    .input
        background-color: #fafafb
        border: 1px solid #eaeaea
        height: 40px
        border-radius: 5px
        text-indent: 15px

        &:focus
            outline: none
            border: 1px solid #615cff7c

    [type=date]
        padding: 0px 15px
        text-indent: 0
