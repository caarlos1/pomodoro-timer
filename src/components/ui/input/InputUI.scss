.input__container {
  display: flex;
  flex-direction: column;
  width: 100%;

  label {
    font-size: 1.4rem;
    margin-bottom: 5px;
    font-weight: 500;
  }

  input {
    background-color: #fff;
    border: 1px solid #eaeaea;
    height: 40px;
    text-indent: 15px;

    &.lg {
      height: 50px;
      font-size: 1.6rem;
      line-height: 1.6rem;
    }

    &.rounded {
      border-radius: 25px;
    }

    &:focus {
      outline: none;
      border: 1px solid #615cff7c;
    }
  }

  [type="date"],
  [type="time"] {
    padding: 0px 15px;
    text-indent: 0;
  }

  [type="number"] {
    padding: 0px 15px;
    text-indent: 0;
  }
}
