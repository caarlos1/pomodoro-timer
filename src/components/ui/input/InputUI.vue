<script setup lang="ts">
import { computed } from "vue";

const props = defineProps({
  modelValue: [String, Number],
  id: String,
  label: String,
  type: {
    type: String,
    default: "text",
  },
});

const emit = defineEmits(["update:modelValue"]);
const inputValue = computed({
  get() {
    return props.modelValue;
  },
  set(value) {
    emit("update:modelValue", value);
  },
});
</script>

<template>
  <div class="input-ui">
    <label v-if="props.label" :for="props.id" class="label">{{
      props.label
    }}</label>
    <input class="input" :id="props.id" :type="type" v-model="inputValue" />
  </div>
</template>

<style lang="sass" scoped>
@import "./InputUI.sass"
</style>
