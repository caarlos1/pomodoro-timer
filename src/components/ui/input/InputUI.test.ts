import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

import InputUI from "./InputUI.vue";

const mock = { id: 1, label: "Little Input" };

describe("InputUI", () => {
  it("should render a input with label", () => {
    const w = mount(InputUI, { props: { label: mock.label } });
    expect(w.get("label").text()).toContain(mock.label);
  });

  it("should emit a event when change input value", () => {
    const w = mount(InputUI, { props: { label: mock.label } });

    const input = w.get("input");
    const inputValue = "Input Value";
    input.setValue(inputValue);

    expect(input.element.value).toBe(inputValue);
    expect(w.emitted()).toHaveProperty("update:modelValue");
  });

  it("should match a snapshot", () => {
    const w = mount(InputUI, { props: { label: mock.label } });
    expect(w.element).toMatchSnapshot();
  });
});
