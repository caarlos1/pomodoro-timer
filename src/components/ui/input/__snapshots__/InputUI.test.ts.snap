// Vitest Snapshot v1

exports[`InputUI > should match a snapshot 1`] = `
<div
  class="input__container"
  data-v-fd244164=""
>
  <label
    data-v-fd244164=""
  >
    Little Input
  </label>
  <input
    class=""
    data-v-fd244164=""
    name="text"
    placeholder=""
    type="text"
  />
</div>
`;
